### DISCUSSION TITLE
redirect fetch utility and passing data

### DISCUSSION THREAD

**Discussion by karkiahmedraid** [remix-run/remix#9388] [Q&A]
is **redirect** utility specified for action and loader function or possible to use it inside react component ? 
another Q how to get data loader of the parent route **or** how to pass data to child route to prevent make another request to server?

**Comment by suhaotian** [comment]
1. yes, redirect utility specified for action and loader function;
2. for global share data, you can put data in the root.tsx;
3. for others use case, check these tips: [Make your SSR(Server-side Rendering) app more stable and faster](https://github.com/suhaotian/xior?tab=readme-ov-file#tips-make-your-ssrserver-side-rendering-app-more-stable-and-faster)
